### PR TITLE
Nds 546 code list component

### PR DESCRIPTION
**Pull request by nikola-nulogy**
### What I did

- Created List and List Item components component
- Added List and updated Typography/Article story to storybook

### How to test

- Run LIst and Article stories in storybook

### CODE REVIEW

File: components/src/List/List.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import styled from "styled-components";
+import {
+  space, color, fontSize, fontWeight, lineHeight,
+} from "styled-system";
+import theme from "../theme";
+import ListItem from "./ListItem";
+
+const List = styled.ul`
+  list-style: none;
+  -webkit-font-smoothing: antialiased;
```
Comment (clemensp) [inline components/src/List/List.js]: Curious about `-webkit` style attributes in general: Should we be adding the attributes that target the other browsers as well?

File: components/src/List/List.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  margin: 0;
+  padding-left: ${theme.space[4]};
+  ${space}
+  ${color}
+  ${fontSize}
+  ${fontWeight}
+  ${lineHeight}
+
+  ${ListItem} {
+    margin-bottom: ${props => (props.compact ? "0" : props.theme.space[2])};
```
Comment (clemensp) [inline components/src/List/List.js]: This can be shortened a bit with: `${({compact, theme}) => `

File: components/src/List/List.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+List.propTypes = {
+  ...space.propTypes,
+  ...color.propTypes,
+  spacing: PropTypes.bool,
+};
+
+List.defaultProps = {
+  mb: 4,
+  color: "inheretColor",
```
Comment (clemensp) [inline components/src/List/List.js]: inheret -> inherit?

Also, is this different from `currentColor`?

File: components/src/List/List.story.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import React from "react";
+import { storiesOf } from "@storybook/react";
+import List from "./List";
+import ListItem from "./ListItem";
+
+storiesOf("List", module)
+  .add("List", () => (
+    <React.Fragment>
```
Comment (clemensp) [inline components/src/List/List.story.js]: Don't need `React.Fragment` if there is only one top-level node.

Comment (clemensp) [reply components/src/List/List.story.js]: Same for the stories below.

File: components/src/Validation/InlineValidation.story.js [hunk 5]
```diff
@@ -1,23 +1,26 @@
...
 
 storiesOf("Inline Validation", module)
   .add("Inline Validation", () => (
     <InlineValidation message="Something has gone wrong">
-      <ul style={ { margin: 0, WebkitFontSmoothing: "antialiased" } }>
-        <li>Something has gone wrong.</li>
-        <li>Entry must be atleast 3 characters long</li>
-        <li><a href="https://nulogy.design/">See here</a></li>
-      </ul>
+      <List compact>
```
Comment (clemensp) [inline components/src/Validation/InlineValidation.story.js:11]: Nice. 😃

**Review by clemensp — CHANGES_REQUESTED** [review]
Some questions to address, but the component looks nice. 👍

File: components/src/List/List.js (hunk 1 above)
Comment (matthewlyle) [reply components/src/List/List.js]: The firefox fontsmoothing is `-moz-osx-font-smoothing: grayscale;`

I wonder if we should be applying this globally though... it's not really a decision unique to List, rather we want all of our fonts to be antialiased. 

We could add it to `ThemeProvider.js` and have it come along when we set the font to Plex.

Comment (nikola-nulogy) [reply components/src/List/List.js]: I agree we should have this applied globally.

Comment (clemensp) [reply components/src/List/List.js]: Yeah. There's probably a library that provides these.
They usually come in sets of 3, right?

Comment (coostenbrug) [reply components/src/List/List.js]: Put this on our ThemeProvider. Ideally we can use a library to handle these attributes on different browsers so I made a new ticket to handle that.

File: components/src/List/List.js (hunk 2 above)
Comment (coostenbrug) [reply components/src/List/List.js]: theme was imported already so I just removed props in props.theme.space[2]

File: components/src/List/List.js (hunk 3 above)
Comment (coostenbrug) [reply components/src/List/List.js]: changed to currentColor

File: components/src/List/List.story.js (hunk 4 above)
Comment (coostenbrug) [reply components/src/List/List.story.js]: fixed

**Review by clemensp — APPROVED** [review]
LGTM. 👍

**Pull request merged**